### PR TITLE
"feat(developer experience): add a dev server

### PR DESCRIPTION
**Pull request by YogliB**
- add a root component  for previewing components
- change `rollup` config to enable "app mode"
- update `README`
- add required node modules

Closes 3"

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width,initial-scale=1" />
+
+    <title>Svelte app</title>
+
+    <link rel="icon" type="image/png" href="/favicon.png" />
+    <link rel="stylesheet" href="/global.css" />
+    <link rel="stylesheet" href="/build/bundle.css" />
```
Comment (shuLhan) [inline public/index.html]: I got 404 on this file when running `npm run dev`. Is bundle.css is not needed or not builded?

Comment (YogliB) [reply public/index.html]: Not needed, I included it by mistake.

Edit: Fixed it.

**Comment by antony** [comment]
Whilst I think this is a decent idea, it makes the template setup considerably more complex than it was. What might be an idea is to investigate and document/provide a setup for something like [svench](https://github.com/rixo/svench) which is probably better suited to component development.

**Comment by shuLhan** [comment]
IMO, as beginner of svelte, the approach suggested by this PR is make sense
and easy to learn for me. User create and test component like building
svelte app (which they should already know), with three simple steps,

1. clone this component repo
2. put component in src/components
3. test it in src/App.svelte

I skim the svench documentation and I immediately dislike it because
another setup/configuration and things to learn (does being "simple" is so
hard in JavaScript module/library?).

Pada tanggal Sab, 22 Agu 2020 04.26, Antony Jones <notifications@github.com>
menulis:

**Comment by antony** [comment]
@shuLhan It's good to strike a balance between keeping the template as basic as possible for people to understand exactly what is required to build svelte components, and muddying the waters by essentially bundling a full application and it's development platform into the component template. I'm sure you can see my POV on this.

Also I'm happy that you skimmed the Svench docs, I don't think theres's *that* much setup, but ideally it would be set up inside the project already and pre-configured, as an isolated platform for live reloading would avoid mixing the concerns of this PR and the template's goals  together.

**Pull request closed without merging**